corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js

```diff
@@ -459,6 +459,15 @@ hqDefine("cloudcare/js/form_entry/fullform-ui", function () {
             $.publish('formplayer.dirty');
         };
 
+        self.getTranslation = function (translationKey, defaultTranslation) {
+            if (self.parent.translations) {
+                var addNewRepeatTranslation = ko.toJS(self.parent.translations[translationKey]);
```

Comment:
What's this `ko.toJS` doing? Isn't `self.parent.translations[translationKey]` already a string?

Reply:
Self.parent is actually a[ knockout object](https://github.com/dimagi/commcare-hq/blob/04824253748a51a7176b1c61ad833ccf97cb03f2/corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js#L170) and so `self.parent.translations[translationKey]` is like a knockout representation and had to be converted back. Perhaps there's a better way to do it? If so, let me merge this so we have it in and then I'll do a quick fix if needed before the deploy.

Reply:
Aha, I see, `self.parent` is a `Form`, which [inherits from Container](https://github.com/dimagi/commcare-hq/blob/04824253748a51a7176b1c61ad833ccf97cb03f2/corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js#L360), which has a [custom function](https://github.com/dimagi/commcare-hq/blob/04824253748a51a7176b1c61ad833ccf97cb03f2/corehq/apps/cloudcare/static/cloudcare/js/form_entry/fullform-ui.js#L150) to turn it into a knockout object. That doesn't have any special handling for `translations`, so ultimately when that function calls `ko.mapping.fromJS`, that'll just turn `translations` into an observable. The standard way to get at the value is to call the observable function: `self.parent.translations()[translationKey]` should do what you want. It's a minor change, up to you if it's worth following up on.

Reply:
That makes sense. I'll look into that, I might have to make a change anyway because it seems like this doesn't work when the repeat is in a group, because the parent is a group object. Oy Vey.